features/ultils/world.ts

```diff
@@ -36,6 +42,54 @@ export class OurWorld extends World {
       destFilePath: filePath,
     });
   }
+
+  public getCredentialByAppKey(appKey: string): Credential {
+    const credential = this.credentials?.find((c) => c.key === appKey);
+    if (credential === undefined) {
+      throw new Error(`The credential with app key ${appKey} is not found`);
+    }
+
+    if (!credential.appId) {
+      throw new Error(`The credential with app key ${appKey} has no App ID`);
+    }
+
+    return credential;
+  }
+
+  public getAPITokenByAppAndPermission(
+    appKey: string,
+    permission: {
+      view?: boolean;
+      add?: boolean;
+      update?: boolean;
+      delete?: boolean;
+    },
+  ): string {
+    const credential = this.getCredentialByAppKey(appKey);
+    const apiToken = credential.apiTokens.find((row) =>
+      Object.entries(permission).every(([key, value]) => {
+        if (value === undefined) {
```

Comment:
@mshrtsr 

When `credential === undefined`, `getCredentialByAppKey` will throw an error.
So, there is no situation that `credential === undefined`.

> When permission doesn't have any element, every will return true without iterating the array.
Oh, you're right. In case of empty, it will return TRUE. It's not our expected.
Let's consider your new design below.

Reply:
Oh, that's not what I mean.
I meaned that `every()` will return true without iterating the array, so you don't need to check `value === undefined`.
Sorry, maybe my typo (`credential === undefined`) was confusing.
> Oh, you're right. In case of empty, it will return TRUE. It's not our expected.